src/main/java/me/zeroeightsix/kami/module/modules/player/PacketLogger.kt

```diff
@@ -3,98 +3,342 @@ package me.zeroeightsix.kami.module.modules.player
+                    }
+                    is CPacketPlayerDigging -> {
+                        add(PacketSide.CLIENT, it.packet,
+                            "positionX: ${it.packet.position.x} " +
+                            "positionY: ${it.packet.position.y} " +
+                            "positionZ: ${it.packet.position.z} " +
+                            "facing: ${it.packet.facing} " +
+                            "action: ${it.packet.action} ")
+                    }
+                    is CPacketEntityAction -> {
+                        add(PacketSide.CLIENT, it.packet,
+                            "action: ${it.packet.action} " +
+                            "auxData: ${it.packet.auxData}")
+                    }
+                    is CPacketUseEntity -> {
+                        add(PacketSide.CLIENT, it.packet,
+                            "action: ${it.packet.action} " +
+                            "hand: ${it.packet.hand} " +
+                            "hitVecX: ${it.packet.hitVec.x} " +
+                            "hitVecY: ${it.packet.hitVec.y} " +
+                            "hitVecZ: ${it.packet.hitVec.z}")
+                    }
+                    is CPacketPlayerTryUseItem -> {
+                        add(PacketSide.CLIENT, it.packet,
+                            "hand: ${it.packet.hand}")
+                    }
+                    is CPacketConfirmTeleport -> {
+                        add(PacketSide.CLIENT, it.packet,
+                            "teleportId: ${it.packet.teleportId}")
+                    }
+                    is CPacketChatMessage -> {
+                        if (!ignoreChat) {
+                            add(PacketSide.SERVER, it.packet,
+                                "message: ${it.packet.message}")
+                        }
+                    }
+                    is CPacketKeepAlive -> {
+                        if (!ignoreKeepAlive) {
+                            add(PacketSide.CLIENT, it.packet,
+                                "key: ${it.packet.key}")
+                        }
+                    }
+                    else -> {
+                        if (!ignoreUnknown) {
+                            add(PacketSide.CLIENT, it.packet, "Not Registered in PacketLogger.kt")
+                        }
+                    }
+                }
+            }
         }
     }
 
-    private fun read() {
-        defaultScope.launch(Dispatchers.IO) {
-            lines.clear()
+    private fun write() {
+        val lines = synchronized(this) {
+            val cache = lines
+            lines = ArrayList()
+            cache
+        }
 
+        defaultScope.launch(Dispatchers.IO) {
             try {
-                file.bufferedReader().forEachLine {
-                    lines.add(it)
+                FileWriter("${KamiMod.DIRECTORY}packetLogs/${filename}", true).buffered().use {
+                    for (line in lines) it.write(line)
```

Comment:
why is this writing line by line?? you can write it all at once without killing your drive

Reply:
buffered writer will do it for you anyways

Reply:
oh ok